Fix visibility docs after refactoring (#775)
temporalio/documentation: docs/go/search-apis.md

Search attributes enable complex and business-logic-focused search queries for Workflow Executions.
These are often queried via the Web UI, but you can also query from within your workflow code (as we show below).

> There are many [Search Attributes](/docs/content/what-is-a-search-attribute) that are added to Workflow Executions by default.

Reviewer: ```suggestion
Many [Search Attributes](/docs/content/what-is-a-search-attribute) are added to Workflow Executions by default.
```

But these are necessarily focused on Temporal internal state tracking.

> For more debugging and monitoring, you may wish add your own domain specific Search Attributes (e.g. `customerId` or `numItems`) that may serve as useful search filters.

Reviewer: ```suggestion
For more debugging and monitoring, you might want to add your own domain-specific Search Attributes (such as `customerId` or `numItems`) that can serve as useful search filters.
```
